CI workflows: updating a few workflows to CXX=20

lucbv opened the pull request:
Getting this in our CI ahead of release 5.0 of the ecosystem which will make it a requirement. The compilers for github OSX and Linux actions are recent enough (gcc 13 and clang 15)

lucbv committed:
CI workflows: updating a few worklows to CXX=20

Getting this in our CI ahead of release 5.0 of the
ecosystem which will make it a requirement. The compilers
for github OSX and Linux actions are recent enough (gcc 13 and clang 15)

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

lucbv committed:
CI - BDW -> Host: renaming broadwell to host

This makes more sense in my mind as we can still name the jobs
within this file to describe the arch they run on but as things
evolve we won't need to rename files and also the at2.yml will
not need to change as well (less mistakes hopefully).

Also switchin back to CXX=17 for the gcc 10.2.0 job as the compiler
is too low for Core requirements.

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

lucbv committed:
CI - Host: more unification of the yaml files

Moving SPR into host and switching build to 4.7.01 even if only
for a short while until 5.0.0 gets released...

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

lucbv committed:
untabify...

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

cwpearson reviewed (changes requested):
Remove the `spr.yml` workflow file too? I presume that's where the new job in `host.yml` came from?

lucbv commented:
Correct, I will also unmark "Required" the two jobs that have moved...

lucbv committed:
CI - Host: removing spr.yml

The job in spr.yml is now included in host.yml so no need to keep
the old file around...

Signed-off-by: Luc Berger-Vergiat <lberge@sandia.gov>

ndellingwood commented on .github/workflows/host.yml:
```diff
@@ -176,8 +176,8 @@ jobs:
 
-  PR_BDW_GNU1020_OPENMP_SERIAL_LEFT_OPENBLAS_REL:
-    name: PR_BDW_GNU1020_OPENMP_SERIAL_LEFT_OPENBLAS_REL
+  PR_GNU1020_OPENMP_SERIAL_LEFT_OPENBLAS_REL:
```
Since gcc/10.2.0 isn't supported for c++20 testing should this job just be removed here?

ndellingwood replied:
Or maybe the c++20 option is fine for gcc/10.2, but gcc/10.2 isn't supported for kokkos@5.0

lucbv replied:
I'm thinking we can keep it until 5.0 comes out then it's easy to turn it off : )
